terraform/main.tf

```diff
@@ -0,0 +1,99 @@
+# Configure the Packet Provider.
+provider "packet" {
+  auth_token = var.auth_token
+}
+
+# Declare your project ID
+locals {
+  project_id = "b0fe85c8-bcae-4760-8866-2ba1f7bb47da"
+}
+
+# Create a new VLAN in datacenter "ewr1"
+resource "packet_vlan" "provisioning-vlan" {
+  description = "provisioning-vlan"
+  facility    = "ewr1"
+  project_id  = local.project_id
+}
+
+# Create a device and add it to tf_project_1
+resource "packet_device" "tf-provisioner" {
+  hostname         = "tf-provisioner"
+  plan             = "c2.medium.x86"
+  facilities       = ["ewr1"]
+  operating_system = "ubuntu_18_04"
+  billing_cycle    = "hourly"
+  project_id       = local.project_id
+  network_type     = "hybrid"
+  
+  connection {
+    user = "root"
+    password = packet_device.tf-provisioner.root_password
+    host = packet_device.tf-provisioner.network[0].address
+  }
+  
+  provisioner "file" {
+    source      = "../cmd/tinkerbell/tinkerbell-cli"
+    destination = "/usr/local/bin/tinkerbell"
+  }
+
+  provisioner "file" {
+    source      = "workflow"
+    destination = "/tmp"
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+        "echo \"HOST_IP=${var.provisioner_ip}\" >> /etc/environment",
+        "echo \"IP_CIDR=${var.cidr}\" >> /etc/environment",
+        "echo \"BROAD_IP=${var.last_ip}\" >> /etc/environment",
+        "echo \"NETMASK=${var.netmask}\" >> /etc/environment",
+        "echo \"DOCKER_USER=${var.quay_user}\" >> /etc/environment",
+        "echo \"DOCKER_PASS=${var.quay_pass}\" >> /etc/environment",
+        "echo \"GIT_USER=${var.git_user}\" >> /etc/environment",
+        "echo \"GIT_PASS=${var.git_pass}\" >> /etc/environment",
+        "echo \"TINKERBELL_REGISTRY_USER=${var.private_registry_user}\" >> /etc/environment",
+        "echo \"TINKERBELL_REGISTRY_PASS=${var.private_registry_pass}\" >> /etc/environment",
+        "echo \"TINKERBELL_GRPC_AUTHORITY=\"127.0.0.1:42113\"\" >> /etc/environment",
+        "echo \"TINKERBELL_CERT_URL=\"http://127.0.0.1:42114/cert\"\" >> /etc/environment",
+        "cat /etc/environment"
+        ]
+  }
+  provisioner "remote-exec" {
+    script = "setup.sh"
+  }
+}
+
+# Create a device and add it to tf_project_1
+resource "packet_device" "tf-worker" {
+  hostname         = "tf-worker"
+  plan             = "c2.medium.x86"
```

Comment:
You might wanna change this. Please check once.

Reply:
Right now this is the only machine type I guess which support layer 2 network conversion. I am planning to put it in the input.tf file and let the user decide the machine type once we have support of layer 2 for other machine types as well.